docs/bugs.json

```diff
@@ -1,4 +1,12 @@
 [
+	{
+		"name": "DynamicArrayCleanup",
+		"summary": "When assigning a dynamic-array in storage, whose types are packed into a slot, certain part of slots were not zeroed out.",
+		"description": "Consider a dynamic-array in storage whose base-type can be packed into a single slot, such as `uint128[]`. Let us define its length to be `l`. When this array gets assigned by another dynamic array with a smaller length, say `m`, the slots between `l` and `m` has to be cleaned by zeroing them out. However, this cleaning was not performed properly. Specifically, if `k` is the number of elements that are packed into a single slot, then from slots `m/k + 1` until `l/k`,  only the first packed value was cleaned up. If this array gets resized to a length larger than `m`, the indices corresponding to the unclean parts of the slot contained the old value, instead of 0. The resizing here is performed by assigning to the array `slot` or assigning to the array `length` or by a `push()`.",
```

Comment:
```suggestion
		"description": "Consider a dynamically-sized array in storage whose base-type is small enough such that multiple values can be packed into a single slot, such as `uint128[]`. Let us define its length to be `l`. When this array gets assigned from another array with a smaller length, say `m`, the slots between elements `l` and `m` have to be cleaned by zeroing them out. However, this cleaning was not performed properly. Specifically, only the first packed value was cleaned up. If this array gets resized to a length larger than `m`, the indices corresponding to the unclean parts of the slot contained the original value, instead of 0. The resizing here is performed by assigning to the array `length`, by a `push()` or via inline assembly. You are not affected if you are only using `.push(<arg>)` or if you assign a value (even zero) to the new elements after increasing the length of the array.",
```

Reply:
Should we remove the part "if `k` is the number of elements that are packed into a single slot, then from slots `m/k + 1` until `l/k`"? Because the entire slot corresponding to the index `m` is properly cleaned up.

Reply:
We could say "after the slot corresponding to `m`, only the first...".